pkg/apis/triggers/v1alpha1/trigger_validation.go

```diff
@@ -55,13 +55,25 @@ func (t TriggerSpecTemplate) validate(ctx context.Context) (errs *apis.FieldErro
 		}
 	}
 
+	// Validate only one of Name or Ref is set.
+	if t.Name != "" && t.Ref != nil {
+		errs = errs.Also(apis.ErrMultipleOneOf("template.name", "template.ref"))
+	}
+
+	// Set Ref to Name
+	if t.Name != "" {
+		t.Ref = &t.Name
+	}
+
 	switch {
-	case t.Spec != nil && t.Name != "":
+	case t.Spec != nil && t.Ref != nil:
 		errs = errs.Also(apis.ErrMultipleOneOf("template.spec", "template.name"))
```

Comment:
DO we need to change error message because we have `ref` here instead of `name` :thinking:

Reply:
Indeed we do 😅